File: app/src/main/java/powerball/apps/jacs/powerball/PastNumbers.java
package powerball.apps.jacs.powerball;

import android.content.Context;
import android.net.Uri;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.design.widget.TabLayout;
import android.support.v4.app.Fragment;
import android.support.v4.view.ViewPager;
import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;



public class PastNumbers extends Fragment {
public ViewPager viewPager;
public ViewPagerAdapter adapter;
    public PastNumbers() {
        // Required empty public constructor
    }


    public static PastNumbers newInstance() {
        PastNumbers fragment = new PastNumbers();

        return fragment;
    }

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
         adapter = new ViewPagerAdapter(getChildFragmentManager());
        adapter.addFragment(new PowerballFragment(), getResources().getString(R.string.powerball));
        adapter.addFragment(new MegaMillionsFragment(), getResources().getString(R.string.megamillions));
    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        // Inflate the layout for this fragment
        View view = inflater.inflate(R.layout.fragment_past_numbers, container, false);
        viewPager = (ViewPager) view.findViewById(R.id.pager1);
        viewPager.setAdapter(adapter);
        viewPager.setOffscreenPageLimit(2);
        TabLayout tabLayout = (TabLayout) view.findViewById(R.id.tabs1);
        tabLayout.setupWithViewPager(viewPager);
        return view;
    }



    @Override
    public void onAttach(Context context) {
        super.onAttach(context);

    }

    @Override
    public void onDetach() {
        super.onDetach();

    }

    @Override
    public void onActivityCreated(@Nullable Bundle savedInstanceState) {
        super.onActivityCreated(savedInstanceState);
        Log.d("timer","PastNumbers onactivity created");
    }

    @Override
    public void onStart() {
        super.onStart();



        // Add Fragments to adapter one by one



    }
}

File: app/src/main/java/powerball/apps/jacs/powerball/PastNumbers.kt
/*
 * Author: John Rowan
 * Description: this is a fragment with a viewpager that has a fragment for each the powerball and the megamillions
 * that both deal with the past results of each lottery
 * Anyone may use this file or anything contained in this project for their own personal use.
 */
package powerball.apps.jacs.powerball

import android.content.Context
import android.os.Bundle
import com.google.android.material.tabs.TabLayout
import androidx.fragment.app.Fragment
import androidx.viewpager.widget.ViewPager
import android.util.Log
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup

class PastNumbers : Fragment() {
    var viewPager: ViewPager? = null
    var adapter: ViewPagerAdapter? = null
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        adapter = ViewPagerAdapter(childFragmentManager)
        adapter!!.addFragment(PowerballFragment(), resources.getString(R.string.powerball))
        adapter!!.addFragment(MegaMillionsFragment(), resources.getString(R.string.megamillions))
    }

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?,
                              savedInstanceState: Bundle?): View? {
        // Inflate the layout for this fragment
        val view = inflater.inflate(R.layout.fragment_past_numbers, container, false)
        viewPager = view.findViewById<View>(R.id.pager1) as ViewPager
        viewPager!!.adapter = adapter
        viewPager!!.offscreenPageLimit = 2
        val tabLayout = view.findViewById<View>(R.id.tabs1) as TabLayout
        tabLayout.setupWithViewPager(viewPager)
        return view
    }

    override fun onAttach(context: Context) {
        super.onAttach(context)
    }

    override fun onDetach() {
        super.onDetach()
    }

    override fun onActivityCreated(savedInstanceState: Bundle?) {
        super.onActivityCreated(savedInstanceState)
        Log.d("timer", "PastNumbers onactivity created")
    }

    override fun onStart() {
        super.onStart()


        // Add Fragments to adapter one by one
    }

    companion object {
        fun newInstance(): PastNumbers {
            return PastNumbers()
        }
    }
}